install.sh

```diff
@@ -178,25 +235,217 @@ legacy_transformations_end() {
 
     # Remove legacy registration file (we moved it to another location)
     if [ -f "$INSTALL_DIR/autoupdate/registered" ]; then
-        rm -rf "$INSTALL_DIR/autoupdate"
+        rm -rf "$INSTALL_DIR/autoupdate" || LEGACY_TRANSFORM_FAILURES="true"
+    fi
+
+    legacy_transform_registered_repos || LEGACY_TRANSFORM_FAILURES="true"
+
+    if [ "$LEGACY_TRANSFORM_FAILURES" = "true" ]; then
+        echo "! There were legacy transform errors: check stderr"
+        return 1
+    fi
+
+    return 0
+}
+
+############################################################
+# Transform all comma-delimited global \`githooks.shared\`
+#   values into multiple git config values.
+#
+# Returns:
+#   1 when failed, 0 otherwise
+############################################################
+legacy_transform_split_global_shared_entries() {
+
+    CURRENT_LIST=$(git config --global --get-all githooks.shared)
+
+    # If we have more then one of these githooks.shared values
+    # we have surely passed this install update already.
+    if [ "$(echo "$CURRENT_LIST" | wc -l)" != "1" ]; then
+        return 0
     fi
+
+    FAILURE="false"
+
+    # If it contains a comma, split it...
+    if echo "$CURRENT_LIST" | grep -q ","; then
+
+        git config --global --unset githooks.shared
+
+        # Split it and add all back
+        IFS="$IFS_COMMA_NEWLINE"
+        for ITEM in $CURRENT_LIST; do
+            unset IFS
+
+            if [ -n "$ITEM" ]; then
+                git config --global --add githooks.shared "$ITEM" || FAILURE="true"
+            fi
+
+            IFS="$IFS_COMMA_NEWLINE"
+        done
+        unset IFS
+    fi
+
+    if [ "$FAILURE" = "true" ]; then
+        echo "! Warning: Could not migrate the global shared hook repositories setting:" >&2
+        echo "\`$CURRENT_LIST\`" >&2
+        echo " Please check \`githooks.shared\` and add all comma-speparated" >&2
+        echo " values manually by running:" >&2
+        echo "  \$ git config --global --add githooks.shared <value>" >&2
+        LEGACY_TRANSFORM_FAILURES="true"
+    fi
+
     return 0
 }
 
 ############################################################
 # Function to dispatch to all legacy transformations
-#   at the end of a install into a repository
+#   for all registered repos.
 #
 # Returns:
 #   1 when failed, 0 otherwise
 ############################################################
-legacy_transformations_repo_end() {
+legacy_transform_registered_repos() {
+
+    # Check if we need transforms for `.shared` files:
+    # Put local paths into `--local githooks.shared`.
+    # which was introduced by PR #125 right after commit ab86d2a5:
+    MOVE_PATHS="false"
+    if legacy_transform_is_ancestor \
+        "$INTERNAL_UPDATED_FROM_COMMIT" \
+        "ab86d2a529f58744a71e79227e434f19b84589e6"; then
+        MOVE_PATHS="true"
+    fi
+
+    if [ "$(git config --global githooks.useCoreHooksPath)" = "true" ]; then
```

Comment:
Hm, what does core.hooksPath have to do with this transformation? Why does it need special treatment?

Reply:
It needs special treatment since in this installation mode we have no way of knowing the registered repos, because we only register when we installed the wrappers (or the wrapper is executed and not registered).
So we just show a warning